### PR TITLE
gron: Read from stdin if no file is provided

### LINKED ISSUE #7870: Utilities: Make gron(1) read from stdin if no file is provided
Currently, `gron(1)` can only read from a file. We should make it able to read from standard input if no file is provided, as most other utilities do. While doing this, it would also make sense to convert `gron` to use `Core::ArgsParser`.

For a simple example of how this could look, take a look at `hexdump.cpp`.

### PR DESCRIPTION
**Pull request by AtkinsSJ**
fixes #7870 

Apologies in advance for anything I'm messing up. This is my first attempt at a pull request, and my git abilities are very rusty.

I noticed while doing this that gron calls VERIFY() to make sure that the Json was valid, which means it crashes if it's not. Would printing an error and returning 1 be better in that case?

### CODE REVIEW

File: Userland/Utilities/gron.cpp [hunk 1]
```diff
@@ -41,15 +42,23 @@ int main(int argc, char** argv)
...
-    if (argc != 2 || !strcmp(argv[1], "--help")) {
-        warnln("usage: gron <file>");
-        warnln("Print each value in a JSON file with its fully expanded key.");
-        return argc != 2 ? 1 : 0;
-    }
-    auto file = Core::File::construct(argv[1]);
-    if (!file->open(Core::OpenMode::ReadOnly)) {
-        warnln("Failed to open {}: {}", file->name(), file->error_string());
-        return 1;
+    Core::ArgsParser args_parser;
```
Comment (MaxWipfli) [inline Userland/Utilities/gron.cpp:45]: You could also add the following here, so we don't remove the informational line above:
```cpp
args_parser.set_general_help("Print each value in a JSON file with its fully expanded key.");
```

**Review by MaxWipfli — COMMENTED** [review]
This looks pretty good. You are right about printing an error, this would definitely make more sense than `VERIFY`.

**Comment by AtkinsSJ** [comment]
Thanks Max! Will do.

Am I right in thinking the `VERIFY` changes should be a separate PR?

**Comment by MaxWipfli** [comment]
> Thanks Max! Will do.
> 
> Am I right in thinking the `VERIFY` changes should be a separate PR?

If you have a couple changes regarding the same program, library or so, you can put them into a single PR if you want, especially if the changes are quite simple. This also makes it a bit easier for the reviewers, I think.

Just make sure to make it a different commit, obviously.

And please squash fixup commits into the original commit, so your "Re-add general help message" commit should be squashed into the first commit.

**Comment by AtkinsSJ** [comment]
Sure. I believe that's everything sorted now.

**Pull request merged**